De-emphasize graceful degradation via removing degraded UI (#618)
primer/design: content/ui-patterns/degraded-experiences.mdx

## Degraded page content

Before making a decision about how to handle UI with unavailable content, check if it already has guidelines on handling errors or empty states.

### Replacing UI with a message

> We don't want users to think they've suffered data loss. If we know a user created something, don't show a generic "empty state". It's better to explain that it's unavailable, or remove the entire section from the page (including the section's heading). For example, I might think my repositories were deleted if I'm on an organization page (like [https://github.com/primer/](https://github.com/primer/)) and the repositories aren't listed there.

Reviewer: ```suggestion
We don't want users to think they've suffered data loss. Instead of a generic "empty state, it is better to explain that it's unavailable, or remove the entire section from the page (including the section's heading). For example, as a user, I might think my repositories were deleted if I am on an organization page (like [https://github.com/primer/](https://github.com/primer/)) and the repositories aren't listed there.
```

  src="https://github.com/primer/design/assets/2313998/bde4367e-8881-41f5-9d06-a2ea5cd79fd6"
/>

### Removing UI

> Be careful about choosing to remove affected UI without explanation. This could cause users to be confused about whether that part of the UI is hidden due to an error, or if it's been permanently removed.

Reviewer: ```suggestion
Be mindful when choosing to remove affected UI without explanation, as this could cause confusion for users about whether that part of the UI is hidden due to an error, or if it's been permanently removed.
```

> However, you may remove UI that's not critical to core workflows and it wouldn't be confusing to render the rest of the page without that UI. For example: it's ok to hide unavailable activity indicators.

Reviewer: ```suggestion
However, it is acceptable to remove UI that's not critical to core workflows and render the rest of the page without that UI. For example: choosing to hide unavailable activity indicators.
```

> Examples of UI that would be disorienting to remove:

Reviewer: ```suggestion
Examples of UI that we do not recommend removing:
```
